Added support for Teensy 3.x and made Wire port selectable in setup

mjs513 opened the pull request:
-  Scope of Change:
Added support for Teensy 3.x series of microcontrollers - excluded pgmspace.h if TEENSYDUINO is used.  Additionally, added the compatibility to select which Wire port you would like to use with the sensor.  For instance:
1.  if you use tsl.begin() in the example it will default to Wire or you can also use tsl.begin(&Wire) but not necessary,
2.  If you want to use Wire2 (Teensy 3.5/3.6 have Wire, Wire1, Wire2 that you can use.

- Limitation:
The only limitation is that there is no default test if you select a Wire port that is not available.

- Test:
Ran a test on the Teensy 3.5 with the sensor on Wire2 and worked fine.

Thanks for considering this change
Mike

ladyada commented:
hiya please remove example where you change Wire2, the examples should not change behavior :)

have you tested with arduino UNO?

mjs513 commented:
I reverted to tsl.begin in the example and added a comment on how to use for other Wire buses. I did not connect the sensor up but I did a compile with no errors for begin() and begin(&Wire).  I can dig it out to give it a try.

ladyada commented:
please do :) it will help us a lot!

mjs513 commented:
Ok.  Tested with Uno, Mega and Due using &Wire1.  Worked fine.

ladyada commented:
thank you kindly :)

mjs513 commented:
Thanks for merging.  Mike
